primefaces/src/main/java/org/primefaces/component/treetable/TreeTable.java

```diff
@@ -588,4 +588,33 @@ protected boolean isCacheableColumns(List<UIColumn> columns) {
         return getFacesContext().getCurrentPhaseId() == PhaseId.RENDER_RESPONSE
                 && (!ComponentUtils.isNestedWithinIterator(this) || columns.stream().noneMatch(DynamicColumn.class::isInstance));
     }
+
+    @Override
+    protected void processNode(FacesContext context, PhaseId phaseId, TreeNode root, TreeNode treeNode, String rowKey) {
+        if (!isPaginator() || root != treeNode) {
+            super.processNode(context, phaseId, root, treeNode, rowKey);
+        }
+        else {
+            if (treeNode != null && shouldVisitNode(treeNode) && treeNode.getChildCount() > 0) {
+                int first = getFirst();
+                int rows = getRows() == 0 ? getRowCount() : getRows();
+
+                processColumnChildren(context, phaseId, root, rowKey);
+
+                List<TreeNode> children = root.getChildren();
+                int childCount = root.getChildCount();
+                int last = (first + rows);
+                if (last > childCount) {
+                    last = childCount;
+                }
+
+                for (int i = first; i < last; i++) {
+                    TreeNode child = children.get(i);
+                    String childRowKey = rowKey == null ? String.valueOf(i) : rowKey + SEPARATOR + i;
```

Comment:
Can we introduce an util method to create the childRowKey, I saw it several times in UITree already

Reply:
I assume you want this util method in the UITree itself?

Reply:
Yes